scripts/components/pixel-canvas.js

```diff
@@ -35,6 +35,56 @@ class PixelCanvasProto extends HTMLElement {
   }
 
   /**
+    Fetch the current canvas state and turn it into a
+    JSON string representing the canvas.
+
+    First we turn the data into a 2D array where each
+    entry is the color hash of the pixels current
+    background color. Then we JSON.stringify that array.
+
+    @return {string} - a string produced by calling JSON.strigify on
+      a 2D Array containing color hex values, such as '#ffffff' for white.
+  */
+  serializeGrid() {
+    let grid = [];
+
+    for(let h = 0; h < this.height; h++) {
+      let row = [];
+      grid[h] = row;
+
+      for(let w = 0; w < this.width; w++) {
+
+          let offset = 2; // Never look where the grid-lines are drawn
+          let x = (this.pixelSize * w) + offset;
+          let y = (this.pixelSize * h) + offset;
+
+          // TODO: may want to pull this into a function
+          let [red, green, blue] = this.canvasCtx.getImageData(x, y, 1, 1).data;
```

Comment:
I don't think so, the documentation says it will always return a `ImageData` element [docs](https://developer.mozilla.org/en-US/docs/Web/API/ImageData). That ImageData would contain data for 100 pixels instead of the average of those pixels if we asked for a 10x10 square. 

There is an example here of a bigger than 1 pixel request:
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData